keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -0,0 +1,319 @@
+
+* Unit tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/disruption/disruption_test.go
+* Integration tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/test/integration/disruption/disruption_test.go
+* Kubectl
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate/versioned/pdb_test.go
+  tests generation of a PDB objects out of given parameters
+  - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/create/create_pdb_test.go
+  tests creation of PDB objects from cmd parameters
+* Scheduler
+  - https://github.com/kubernetes/kubernetes/blob/4b59044b8d2a3502ea490ba2c958008a098511a3/test/integration/scheduler/preemption_test.go#L1053
+  tests effects of PDB on preemption (PDB is honored in a best effort way)
+* Eviction integration tests
+  - https://github.com/kubernetes/kubernetes/blob/master/test/integration/evictions/evictions_test.go test eviction logic and its interactions with PDB.
+* Autoscaler
+  - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/autoscaling/cluster_size_autoscaling.go ensure that Autoscaler respects PDB when draining nodes.
+* Integration tests for disruption controller
+  - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/disruption.go
+
+All beta PDB endpoints are covered by e2e tests: https://apisnoop.cncf.io/1.20.0/beta/policy
+
+#### Conformance tests
+
+The following e2e tests will be included in the conformance tests:
+
+ * DisruptionController
+   * should block an eviction until the PDB is updated to allow it
+   * create a PodDisruptionBudget
+   * delete collection of PodDisruptionBudget
+   * delete a PodDisruptionBudget
+   * list or watch objects of kind PodDisruptionBudget
+   * partially update the specified PodDisruptionBudget
+   * partially update status of the specified PodDisruptionBudget
+   * read the specified PodDisruptionBudget
+   * read status of the specified PodDisruptionBudget
+   * replace the specified PodDisruptionBudget
+   * replace status of the specified PodDisruptionBudget
+
+### Graduation Criteria
+
+Graduation to GA:
+- [x] Implement Mutable PDBs
+- [ ] Address performance issues
+- [ ] Pass conformance tests
+- [ ] Update documents to reflect the changes
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: PodDisruptionBudget
+    - Components depending on the feature gate: kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  No
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes
+
+* **What happens if we reenable the feature if it was previously rolled back?** 
+  No changes expected
```

Comment:
technically speaking - we will recalculate the PDB statuses again and start using them again.

Reply:
Updated.